### ISSUE TITLE
VSCode's change to Powershell breaks workflow for Elixir

### ISSUE THREAD

**Issue by ghost** [PowerShell/vscode-powershell#520]
(from https://github.com/Microsoft/vscode/issues/21753)
VSCode Versions 1.9 - 2.0
Windows 10

Steps to Reproduce:

1.  Install Elixir following the [official docs}(http://elixir-lang.org/install.html)
2.  Install VSCode version < 1.9
3.  Press ctrl + ` to open VSCode's command line
4.  Elixir REPL can be started with iex and mix tasks work as expected.
5.  Upgrade VSCode and reopen it
6.  Press ctrl + ` to open VSCode's command line
7.  Both iex and mix now fail.

**Comment by rkeithhill** [comment]
In VSCode 1.9 the default "terminal" on Windows has switched from CMD.exe to PowerShell.exe to align with Windows move to make PowerShell the default terminal in the UI.  

In PowerShell, `iex` is an alias so try `iex.exe`.  That should run.  I'm not sure why `mix` would be failing as that is not a PowerShell alias.  Try `mix.exe`.  

Also, you can always switch the terminal to CMD.exe with the following user setting:
```
  "terminal.integrated.shell.windows": "C:\\WINDOWS\\Sysnative\\cmd.exe"
```

**Comment by toshuo** [comment]
See: https://github.com/Microsoft/vscode/issues/21753#issuecomment-283683612

**Comment by rkeithhill** [comment]
I don't believe this issue has anything to do with the PowerShell extension. You are using the default terminal in VSCode which as of 1.9, starts PowerShell.  The issue with `iex` is that `iex` is an alias in PowerShell.  I've shown you how to remove that alias in the VSCode issue thread:
```powershell
Remove-Item Alias:\iex -Force
```
The issue with mix is that the kind Elixir folks have provided a mix.ps1 (PowerShell script) file. Can you tell me what you get back from the command `Get-Command mix`?  I suspect it will refer to the Elixir installed mix.ps1 file.  PowerShell script file execution is disabled by default (anybody remember the ILoveYou VBScript virus?).  
```
> mix
mix : C:\Program因為這個系統上已停用指令碼執行，所以無法載入 C:\Program Files (x86)\Elixir\bin\mix.ps1 檔案。如需
詳http://go.microsoft.com/fwlink/?LinkID=135170細資訊，請參閱 about_Execution_Policies，網址為 http://go.microsoft.com/fwlink/?LinkID=135170。
位於 線路:1 字元:1
```
Anyway, you have two options. You can run mix.bat or you can set the PowerShell execution policy to allow PowerShell scripts to execute.  You can do the latter by opening a PowerShell (x64) console **as admin** and executing `Set-ExecutionPolicy RemoteSigned`.   If that still causes the `UnauthorizedAccess` error when you run `mix`, run the command `Unblock-File <path-mix.ps1>` and try again.

**Comment by toshuo** [comment]
>  Can you tell me what you get back from the command Get-Command mix?

```
> Get-Command mix

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
ExternalScript  mix.ps1                                                       C:\Program Files (x86)\Elixir\bin\mix.ps1
```
> anybody remember the ILoveYou VBScript virus?
No idea what that was... I've never used VB.

> by opening a PowerShell (x64) console **as admin**
How is that done? Can it be done from VSC?

> I don't believe this issue has anything to do with the PowerShell extension.

Maybe not, @rkeithhill. I filed it here because @isidorn told me to (see: https://github.com/Microsoft/vscode/issues/21753#issuecomment-283626094)

Anyway, I don't know where this issue should be filed but I know clearly what the issue is: 
**Visual Studio Code on Windows is now broken for Elixir development by default**.

There are also problems for other top 50 languages. I understand if that's simply not a priority, but the truth is version 1.9+ is hostile to non-windows devs using all but the most popular few languages/stacks. Defaults matter because switching costs are so low. I already abandoned work on my windows machine (at least until there's a patch or simple script to fix the setup) because I had pressing work to do. If too many new users less patient than I am do the same, then plugins will lag further behind IntelliJ, ST3, Atom etc, and VSC becomes a weaker choice.

Anyway, I don't want to press the point further since it looks like the team chose this path intentionally. I was just reporting the issue.

**Comment by rkeithhill** [comment]
> How is that done? (run as admin)

Press the Windows Start button, type PowerShell, when it appears at the top of the list, right click it and select "Run as administrator":
![image](https://cloud.githubusercontent.com/assets/5177512/23621162/d4fee30c-0256-11e7-813c-e07bf6f7201f.png)
Then run the command `Set-ExecutionPolicy RemoteSigned`.  Verify that `mix` executes.  Now you should be able to run `mix` from the VSCode PowerShell terminal.

**Comment by daviwil** [comment]
With VS Code 1.11, you are now prompted to configure your default terminal the first time one opens.  If you select cmd.exe in this popup it should fix the problems you were having by the change in default to Windows PowerShell.
